// zencf/index.mjs
import fetch from 'node-fetch'

const API = 'https://cf.zenzxz.web.id/solve'

async function request(payload) {
  const res = await fetch(API, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload)
  })

  const json = await res.json()
  if (!json.status) throw new Error(json.error || 'Request failed')

  const data = json.data

  if (data?.headers) {
    data.headers['User-Agent'] =
      data.headers['User-Agent'] || data.headers['user-agent']
  }

  return data
}

function parseProxy(proxy) {
  if (!proxy) return undefined

  const [auth, hostport] = proxy.includes('@')
    ? proxy.split('@')
    : [null, proxy]

  if (!hostport || !hostport.includes(':')) return undefined

  const [host, port] = hostport.split(':')
  if (!host || !port || isNaN(port)) return undefined

  if (auth) {
    const [username, password] = auth.split(':')
    return { host, port: Number(port), username, password }
  }

  return { host, port: Number(port) }
}

export const zencf = {

  async wafSession(url, proxy) {
    if (!url) throw new Error('url wajib')

    return request({
      url,
      mode: 'waf-session',
      proxy: parseProxy(proxy)
    })
  },

  async turnstileMin(url, siteKey, proxy) {
    if (!url || !siteKey) throw new Error('url dan siteKey wajib')

    const r = await request({
      url,
      siteKey,
      mode: 'turnstile-min',
      proxy: parseProxy(proxy)
    })

    return { token: r.token || r.value || r }
  },

  async turnstileMax(url, proxy) {
    if (!url) throw new Error('url wajib')

    const r = await request({
      url,
      mode: 'turnstile-max',
      proxy: parseProxy(proxy)
    })

    return { token: r.token || r.value || r }
  },

  async source(url, proxy) {
    if (!url) throw new Error('url wajib')

    return request({
      url,
      mode: 'source',
      proxy: parseProxy(proxy)
    })
  }

}

export default { zencf }
