// genshin-api/dist/index.js
const fetch = require("node-fetch");
const Artifact = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/artifacts/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Characters = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/characters/${type}`).then(
      (body) => body.json()
    );
    data.image = `https://genshin.jmp.blue/characters/${type}/portrait`;
    resolve(data);
  });
  return promise;
};

const Domains = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/domains/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Elements = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/elements/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Enemies = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/enemies/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Nations = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/nations/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

const Weapons = (type) => {
  if (type == undefined) {
    type = "";
  }
  var promise = new Promise(async function (resolve, reject) {
    let data = await fetch(`https://genshin.jmp.blue/weapons/${type}`).then(
      (body) => body.json()
    );

    resolve(data);
  });
  return promise;
};

exports.Artifact = Artifact;
exports.Characters = Characters;
exports.Domains = Domains;
exports.Elements = Elements;
exports.Enemies = Enemies;
exports.Nations = Nations;
exports.Weapons = Weapons;
